android/src/main/kotlin/com/airship/flutter/ConfigManager.kt

```diff
@@ -26,42 +28,174 @@ class ConfigManager(private val context: Context) {
 
             val config = AirshipConfigOptions.newBuilder()
-                .applyDefaultProperties(context)
+                //.applyDefaultProperties(context)
+                .setRequireInitialRemoteConfigEnabled(true)
                 .apply {
-                    if (!(appCredentialsOverride.first.isNullOrEmpty() || appCredentialsOverride.second.isNullOrEmpty())) {
-                        this.setAppKey(appCredentialsOverride.first)
-                            .setAppSecret(appCredentialsOverride.second)
+                    pluginConfig.inProduction.let {
+                        println(it)
+                        this.setInProduction(it)
+                    }
+
+                    pluginConfig.development.let {
+                        this.setDevelopmentAppKey(it.appKey)
+                            .setDevelopmentAppSecret(it.appSecret)
+                            .setDevelopmentLogLevel(it.logLevel.value)
+                            .build()
+                    }
+
+                    pluginConfig.production.let {
+                        this.setProductionAppKey(it.appKey)
+                            .setProductionAppSecret(it.appSecret)
+                            .setProductionLogLevel(it.logLevel.value)
+                            .build()
+                    }
+
+                    /// since proto buf defaults to empty string,
+                    // check for empty string instead of null
+                    pluginConfig.defaultEnv.let {
+                        this.setAppKey(it.appKey)
+                            .setAppSecret(it.appSecret)
+                            .setLogLevel(it.logLevel.value)
                             .build()
                     }
+
+
+                    this.setSite(
+                        when (pluginConfig.site) {
+                            Config.Site.SITE_EU -> {
+                                AirshipConfigOptions.SITE_EU
+                            }
+                            Config.Site.SITE_US -> {
+                                AirshipConfigOptions.SITE_US
+                            }
+                            Config.Site.UNRECOGNIZED -> {
+                                AirshipConfigOptions.SITE_US
+                            }
+                            else-> AirshipConfigOptions.SITE_US
+                        }
+                    )
+
+
+
+                    pluginConfig.urlAllowListList.let {
+                        this.setUrlAllowList(null/*it.toTypedArray()*/)
+                    }
+
+                    pluginConfig.urlAllowlistScopeJavascriptInterfaceList.let {
+                        this.setUrlAllowListScopeJavaScriptInterface(it.toTypedArray())
+                    }
+
+                    pluginConfig.urlAllowListScopeOpenUrlList.let {
+
+                        this.setUrlAllowListScopeOpenUrl(it.toTypedArray())
+                    }
+
+                    this.setEnabledFeatures(pluginConfig.featuresEnabledList.supportValue())
```

Comment:
I think we may want something like this, unless the default of `FEATURE_ALL` is being handled by the Proto definition:

``` kotlin
if (pluginConfig.featuresEnabledCount > 0) {
    config.setEnabledFeatures(pluginConfig.featuresEnabledList.supportValue())
} else {
    config.setEnabledFeatures(PrivacyManager.FEATURE_ALL)
}
```

Reply:
It is done through protobuf